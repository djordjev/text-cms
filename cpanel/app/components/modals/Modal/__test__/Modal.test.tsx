import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { Modal, ModalProps } from '../Modal';

test('Modal', () => {
  const onClose = vi.fn();

  const props: ModalProps = {
    children: <div data-testid="modal_content" />,
    onClose,
    open: true,
    title: 'test_title'
  };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/',
        Component: () => <Modal {...props} />
      }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByText(props.title!)).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
