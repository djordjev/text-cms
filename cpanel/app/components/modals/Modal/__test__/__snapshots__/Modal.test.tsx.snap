// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal 1`] = `
NodeList [
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />,
]
`;
